tools/run_tests/run_tests.py

```diff
@@ -278,7 +278,8 @@ def configure(self, config, args):
             self._cmake_architecture_windows = 'x64' if self.args.arch == 'x64' else 'Win32'
             # when builing with Ninja, the VS common tools need to be activated first
             self._activate_vs_tools_windows = activate_vs_tools
-            self._vs_tools_architecture_windows = 'x64' if self.args.arch == 'x64' else 'x86'
+            # always use x64 toolset to overcome the memory limit of 32bit linker
+            self._vs_tools_architecture_windows = 'x64'
```

Comment:
wouldn't this make 32bit build on windows silently become 64bit? https://github.com/grpc/grpc/blob/0afcc375890eef6de23c8e0a29b6953b3f07b8e0/tools/run_tests/helper_scripts/build_cxx.bat#L31

Let's doublecheck that before proceeding with this PR.

Reply:
Sure! My expectation is that it only controls the tool and generated artifacts should follow `_cmake_architecture_windows` but let's double check.

Reply:
I don't think your assumptions are correct. There are actually 2 ways of building (see the if then else in the script)

1. build with Ninja, which doesn't set architecture through cmake, but instead relies on the default architecture for the vs tools you selected (and I think this is the case that will get broken by your change)
https://github.com/grpc/grpc/blob/0afcc375890eef6de23c8e0a29b6953b3f07b8e0/tools/run_tests/helper_scripts/build_cxx.bat#L41
2. build with cmake's visual studio generator, which lets you pass `-A` architecture flag to cmake
https://github.com/grpc/grpc/blob/0afcc375890eef6de23c8e0a29b6953b3f07b8e0/tools/run_tests/helper_scripts/build_cxx.bat#L48

Reply:
Yeah, `vcvarsall` didn't turn out to be effective. But this change worked. I verified this PR on the personal kokoro image and that's the output.

```
T:\src\github\grpc>file workspace_c++_windows_dbg_native_default_default_buildonly\cmake\build\useful_test.exe
file workspace_c++_windows_dbg_native_default_default_buildonly\cmake\build\useful_test.exe
workspace_c++_windows_dbg_native_default_default_buildonly\cmake\build\useful_test.exe: PE32 executable (console) Intel 80386, for MS Windows

T:\src\github\grpc>file workspace_c++_windows_dbg_native_x64_cmake_ninja_vs2017_buildonly\cmake\build\useful_test.exe
file workspace_c++_windows_dbg_native_x64_cmake_ninja_vs2017_buildonly\cmake\build\useful_test.exe
workspace_c++_windows_dbg_native_x64_cmake_ninja_vs2017_buildonly\cmake\build\useful_test.exe: PE32+ executable (console) x86-64, for MS Windows
```